features/48-user-profile-discovery/field-based-acl.md

```diff
@@ -107,10 +107,50 @@ function recordOperationVeridation (operation) {
 }
 ```
 
+## Partial Record Fields Update
+
+For backward compatibility of Skygear API, partial record fields update is
+introduced. For non-atomic record save request, partial fields update is
+allowed, which means that even some record fields are not writable for the
+requesting user, the record will still be updated on other writable fields.
+An warning message would be responded to developers showing which field updates
+are rejected because of the field-based ACL.
+
 # Changes on SDK
 
-No changes on SDK since this feature would be expected to be used on Skygear
-Cloud Portal or Skygear CLI, which will be specified on another document.
+Skygear SDKs have not many changes since the setup of field-based ACL would be
+expected on Skygear Cloud Portal or Skygear CLI, which will be specified on
+another document.
+
+The record save API on Skygear SDKs would be updated to support partial record
+fields update on non-atomic request.
+
+```js
+// saving one record
+skygear.publicDB.query(record1)
+.then((savedRecord) => {
```

Comment:
I am suggesting `.then((savedRecord) => {` ==> `.then((savedRecord, error) => {`. For array of records, `.then((savedRecords) => {` ==> `.then((savedRecords, errors) => {`

So the code will become follow:

```
then((savedRecord, error) => {
  console.log(`Successfully saved record: ${savedRecord}`);
  console.warn(`Rejected saving fields: ${error.$rejectedFields}`)
}, (err) => {
  console.error(`Failed to save: ${err}`);
});
```

Since `savedRecord` is expected to be an instance of `Record`, adding `$rejectedFields` to the class will add knowledge to the `Record` class. It is not necessary and not good to make the user need to understand field ACL upfront.

Second, we are designing for a smooth normal flow. The API will encourage people to separate normal code and error handling code. And on first draft of their application, the API should not require them to handle/understand every error before they ship a demo.